jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1427,6 +1427,48 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
   return ret_value;
 } /* ecma_regexp_exec_helper */
 
+/**
+ * Helper function for converting a RegExp pattern parameter to string.
+ *
+ * See also:
+ *         RegExp.compile
+ *         RegExp dispatch call
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_regexp_read_pattern_str_helper (ecma_value_t pattern_arg, /**< the RegExp pattern */
+                                     ecma_string_t **pattern_string_p) /**< ptr to store it as string */
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+  if (!ecma_is_value_undefined (pattern_arg))
+  {
+    ECMA_TRY_CATCH (regexp_str_value,
```

Comment:
We've already started the the elimination of `ECMA_TRY_CATCH`. Please avoid the usage of it.